commit 1b031e7d6ba45724d6dfc905867e3e1c915b7a56 (refs/changes/51/2251/1)
Author: David Pacheco <dap@joyent.com>
Date:   2017-07-21T12:13:02-07:00 (2 years, 3 months ago)
    
    joyent/node-manta#244 mlogin could disable Manta's abort-on-core behavior

diff --git a/CHANGES.md b/CHANGES.md
index 305ed35..5ec55a8 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -2,6 +2,10 @@
 
 ## not yet released
 
+## 4.4.3
+
+- joyent/node-manta#244 mlogin could disable Manta's abort-on-core behavior
+
 ## 4.4.2
 
 - joyent/node-manta#312 Custom header input should tolerate ':' characters
diff --git a/bin/mlogin b/bin/mlogin
index ddd87e2..262ec91 100755
--- a/bin/mlogin
+++ b/bin/mlogin
@@ -151,10 +151,17 @@ function parseOptions() {
 
     if (opts.command) {
         opts.arguments = [ '-c', opts.command ];
+        opts.command = '/bin/bash';
     } else {
-        opts.arguments = [ '--norc' ];
+        /*
+         * The default behavior in Manta jobs is that if a process dumps core,
+         * then the whole task is aborted.  That's not appropriate (and pretty
+         * annoying) for interactive use, so we run the command in our own
+         * contract that will ignore when a process dumps core.
+         */
+        opts.command = '/usr/bin/ctrun';
+        opts.arguments = [ '-i', 'core', '-l', 'child', '/bin/bash', '--norc' ];
     }
-    opts.command = '/bin/bash';
 
     opts.running = true;
     opts.cancel_job = false;
diff --git a/package.json b/package.json
index b9dd171..6b7df03 100644
--- a/package.json
+++ b/package.json
@@ -7,7 +7,7 @@
         "type": "git",
         "url": "git://github.com/joyent/node-manta.git"
     },
-    "version": "4.4.2",
+    "version": "4.4.3",
     "main": "./lib/index.js",
     "dependencies": {
         "assert-plus": "^1.0.0",
